packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -98,7 +98,8 @@ export class TelemetryService {
       });
 
     const machineId = env ? env.machineId : 'someValue.machineId';
-    const isDevMode = machineId === 'someValue.machineId';
+    const isDevMode =
+      extensionContext.extensionMode !== ExtensionMode.Production;
```

Comment:
So this wasn't specific to these changes (also started failing in develop). I think the issue is machineId use to not return a value in the test env, but now it does. I updated to use the extensionMode property on the vscode module to determine if we are in dev mode.